scripts/zones/Western_Altepa_Desert/mobs/King_Vinegarroon.lua

```diff
@@ -36,4 +48,18 @@ entity.onMobDespawn = function(mob)
     mob:setRespawnTime(math.random(75600, 86400)) -- 21 to 24 hours
 end
 
+entity.mobRegen = function(mob)
```

Comment:
While you can attach this to the entity object, it's not really a patten we use anywhere else, we'd tend to just make this a local function nearer the top of the file:
```lua
local mobRegen = function(mob)
    local hour = VanadielHour()

    if hour >= 6 and hour <= 20 then
        mob:setMod(xi.mod.REGEN, 125)
    else
        mob:setMod(xi.mod.REGEN, 250)
    end
end

-- lower down 

mobRegen(mob)
```